process/transaction/shardProcess.go

```diff
@@ -530,6 +510,7 @@ func (txProc *txProcessor) processRelayedTx(
 	}
 
 	totalFee, remainingFee, relayerFee, remainingGasLimit := txProc.computeRelayedTxFees(tx)
+	//TODO: is this check really necessary? Why?
```

Comment:
yes it is, in order to correctly send and use gas all over the protocol. to have correct refunds. If this invariant does not stay it is hard to manage the rest. There is no reason for a relayer to pay more gas than this.

Reply:
ok